pkg/recipe/metadata.go

```diff
@@ -54,6 +59,10 @@ func (m *Metadata) Validate() error {
 		return fmt.Errorf("version \"%s\" is not a valid semver", m.Version)
 	}
 
+	if m.TemplateExtension != "" && m.TemplateExtension[0] != '.' {
```

Comment:
`strings.HasPrefix()` could be utilized here
```suggestion
	if m.TemplateExtension != "" && !strings.HasPrefix(m.TemplateExtension, '.') {
```